src/main/java/org/truffleruby/parser/lexer/RubyLexer.java

```diff
@@ -724,13 +721,18 @@ private int hereDocumentIdentifier() {
             indent = Integer.MAX_VALUE;
         }
 
+        int token = tSTRING_BEG;
```

Comment:
I've tried removing all the changes under this function `private int hereDocumentIdentifier()` and it doesn't effect the entire commit. i.e. the test for the original issue is still passing, and the same new test is failing. 

So perhaps the changes here are not required?